spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/StructTypeImplicits.scala

```diff
@@ -417,4 +417,45 @@ object StructTypeImplicits {
     }
   }
 
+  implicit class StructTypeComparisonWhileIgnoringNullability(val schema: StructType) {
+    /**
+     * Wraps an array of StructField with StructType
+     *
+     * @param fieldSchema is the actual schema to be wrapped
+     * @return StructType containing the StructFields
+     */
+    def wrapStructFiledWithStructType(fieldSchema: Option[DataType]): Boolean = {
+      val st = StructType(Seq(StructField("errCol", fieldSchema.headOption.head)))
+      dataTypeEqualsIgnoreNullability(st, schema)
+    }
+
+    /**
+     * Compares two column schemas while ignoring nullability
+     *
+     * @param colSchema1 one of the column schema to be compared
+     * @param colSchema2 another column schema to be compared
+     * @return boolean type for compared column schemas/types
+     */
+    private def dataTypeEqualsIgnoreNullability(colSchema1: DataType, colSchema2: DataType): Boolean = {
+      (colSchema1, colSchema2) match {
+        case (st1: StructType, st2: StructType) =>
+          if (st1.fields.length != st2.fields.length) {
+            false
+          } else {
+            st1.fields.zip(st2.fields).forall {
+              case (f1, f2) =>
+                f1.name == f2.name && dataTypeEqualsIgnoreNullability(f1.dataType, f2.dataType)
+            }
+          }
+
+        case (ArrayType(et1, n1), ArrayType(et2, n2)) =>
+          dataTypeEqualsIgnoreNullability(et1, et2) && n1 == n2
```

Comment:
I really like this function. :) I think it's useful and I like the way you used recursion and pattern matching. 

However, the code docs says: `while ignoring nullability` and those `n1 == n2` comparisons (here and 1 more below) are comparing `containsNull` condition. 

What did you mean by 'ignoring nullability' then?

Reply:
What I mean is the function only compares the field names and their dataTypes. Then it doesn't care if the field cantains null or not.

Reply:
Hmm, but that's my point - it doesn't really only do that, right? So back to my original question, what about those `n1 == n2` ?